homeassistant/components/snapcast/media_player.py

```diff
@@ -275,6 +279,68 @@ async def async_unjoin(self) -> None:
         """Handle the unjoin service."""
         raise NotImplementedError
 
+    def _get_metadata(self, key, default=None) -> Any:
```

Comment:
Would it make sense to create a
```py
@property
def metadata(self) -> ..:
    return self.coordinator.server.stream.....
```

And then just have the properties down below do `self.metadata.get("title")`?

Reply:
Alright, I made the property fallback to returning an empty dict so the other metadata properties don't have to check for None.